src/ott/geometry/costs.py

```diff
@@ -239,9 +240,10 @@ def fun(z: jnp.ndarray, x: jnp.ndarray) -> float:
 
     def f_h(x: jnp.ndarray) -> float:
       solver = jaxopt.LBFGS(fun=fun, **kwargs)
-      solver = solver.run(x, x=x)
-      sol = jax.lax.stop_gradient(solver.params)
-      return fun(sol, x)
+      x0 = x if x_init is None else x_init
```

Comment:
can you already embed the option of passing a Sequence of x0 instead of just one and then taking a min over the results?